python/tests/server/test_worker.py

```diff
@@ -465,42 +498,57 @@ def __init__(self):
         self.predict_generator = None
         self.predict_events = []
         self.predict_payload = None
+        self.setup_done = False
+
+    def await_(self, coro: Awaitable[T]) -> T:
+        return self.loop.run_until_complete(coro)
 
     @rule(sleep=st.floats(min_value=0, max_value=0.5))
     def wait(self, sleep):
         time.sleep(sleep)
 
+    @precondition(lambda x: not x.setup_done)
     @rule()
     def setup(self):
+        trace("did setup")
         try:
             self.setup_generator = self.worker.setup()
             self.setup_events = []
+        # this except allows setup to be called many times which seems weird
```

Comment:
The point of the Hypothesis test is to ensure that calling the methods in any order "works," where "works" might mean throwing `InvalidStateException` because it's not a reasonable time to call that method. Adding a `precondition` and removing this removes that validation.

Reply:
is there some actual point to testing multiple setup() like that? the state error only happens when you try to consume the stream. if you call setup or predict multiple times without actually consuming the generator there's no reason it should work (it will never get to the assertion)